### PR TITLE
Add delivery fees

### PR DESCRIPTION
**Pull request by girazoki**
Add delivery fees for XCM routers in all runtimes (except Flashbox). We use `ExpontentialPrice` from polkadot-runtime-common and we charge in the native asset for each runtime.

Added a couple function in TS so that we can check that indeed the messages are correctly charged. These take the latest fees from the sent messages available in storage

### CODE REVIEW

File: runtime/dancebox/src/xcm_config.rs [hunk 1]
```diff
@@ -245,14 +249,20 @@ impl pallet_xcm::Config for Runtime {
     type AdminOrigin = EnsureRoot<AccountId>;
 }
 
+pub type PriceForSiblingParachainDelivery =
+    ExponentialPrice<SelfReserve, BaseDeliveryFee, TransactionByteFee, XcmpQueue>;
+
+pub type PriceForParentDelivery =
+    ExponentialPrice<FeeAssetId, BaseDeliveryFee, TransactionByteFee, ParachainSystem>;
```
Comment (Agusrodri) [inline runtime/dancebox/src/xcm_config.rs:256]: Why do we need the conversion to `FeeAssetId` type? Couldn't we use the same `SelfReserve` as in `PriceForSiblingParachainDelivery`?

Comment (girazoki) [reply runtime/dancebox/src/xcm_config.rs:256]: yep I think so

File: test/util/xcm.ts [hunk 2]
```diff
@@ -919,3 +919,43 @@ export const expectXcmEventMessage = async (context: DevModeContext, message: st
...
+    const records = await context.polkadotJs().query.system.events();
+
+    const filteredEvents = records
+        .map(({ event }) => (context.polkadotJs().events.polkadotXcm.FeesPaid.is(event) ? event : undefined))
+        .filter((event) => event);
+
+    return filteredEvents[0]!.data[1][0].fun.asFungible.toBigInt();
+};
+
+export const getLastSentUmpMessageFee = async (context: DevModeContext, baseDelivery: bigint, txByteFee: bigint) => {
```
Comment (Agusrodri) [inline test/util/xcm.ts:933]: Really useful functions :)

**Review by Agusrodri — APPROVED** [review]
Looks great! Specially the typescript helpers/tests addition 🥇

**Review by fgamundi — APPROVED** [review]

File: test/util/xcm.ts [hunk 3]
```diff
@@ -919,3 +919,43 @@ export const expectXcmEventMessage = async (context: DevModeContext, message: st
...
+    const sentXcm = (await context.polkadotJs().query.parachainSystem.upwardMessages())[0];
```
Comment (ParthDesai) [inline test/util/xcm.ts:934]: Could this be a runtime error if the message list is empty?

**Review by ParthDesai — APPROVED** [review]
Minor comment in test. Apart from that everything else look good.

File: test/util/xcm.ts (hunk 3 above)
Comment (girazoki) [reply test/util/xcm.ts:934]: If the message list is empty it will simply throw a typescript error.

Comment (girazoki) [reply test/util/xcm.ts:934]: the runtime will return an empty vec and you will try to get the item 0, which does not exist

Comment (ParthDesai) [reply test/util/xcm.ts:934]: yes, I mean typescript error. Do we safeguard against that? Or it is fine?

**Pull request merged**